modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala

```diff
@@ -420,6 +420,18 @@ object EpicsUtil {
   def smartSetParam[A: Eq](v: A, get: => Option[A], set: SeqAction[Unit]): List[SeqAction[Unit]] =
     if(get =!= v.some) List(set) else Nil
 
+  def applyParam[F[_]: Applicative, A: Eq](c: A, d: A, f: A => F[Unit]): Option[F[Unit]] =
+    if (c =!= d) f(d).some else none
+
+  def applyParamT[F[_]: Functor](
+    relTolerance: Double
+  )(c: Double, d: Double, set: Double => F[Unit]): Option[F[Unit]] =
+    if (!(d === 0.0 && c === 0.0) || (d === 0.0 || abs((c - d)/d) > relTolerance)) {
+      set(d).some
```

Comment:
!(d === 0.0 && c === 0.0) **&&** (d === 0.0 || abs((c - d)/d) > relTolerance)
If current and demanded value are both 0.0, then we don't need to set it.